[Bug][Alert] ERROR: relation "t_ds_plugin_define" does not exist

**To Reproduce**
Steps to reproduce the behavior, for example:
1. Run `docker-compose up -d`
2. Run `docker logs -f dolphinscheduler-alert`
3. See error

**Expected behavior**
Bug fixed

**Screenshots**
Alert logs:
```
[INFO] 2021-04-26 00:27:39.678 org.apache.dolphinscheduler.alert.plugin.AlertPluginManager:[104] - Registering Alert Plugin 'DingTalk'
[INFO] 2021-04-26 00:27:39.681 org.apache.dolphinscheduler.alert.plugin.AlertPluginManager:[86] - -- Loaded Alert Plugin DingTalk --
[INFO] 2021-04-26 00:27:41.901 com.alibaba.druid.pool.DruidDataSource:[994] - {dataSource-1} inited
Exception in thread "main" java.lang.RuntimeException: load Alert Plugin Failed !
	at org.apache.dolphinscheduler.alert.AlertServer.initPlugin(AlertServer.java:104)
	at org.apache.dolphinscheduler.alert.AlertServer.start(AlertServer.java:145)
	at org.apache.dolphinscheduler.alert.AlertServer.main(AlertServer.java:161)
Caused by: org.springframework.jdbc.BadSqlGrammarException: 
### Error querying database.  Cause: org.postgresql.util.PSQLException: ERROR: relation "t_ds_plugin_define" does not exist
  Position: 23
### The error may exist in org/apache/dolphinscheduler/dao/mapper/PluginDefineMapper.xml
### The error may involve defaultParameterMap
### The error occurred while setting parameters
### SQL: select *         from t_ds_plugin_define         where plugin_name = ? and plugin_type = ?
### Cause: org.postgresql.util.PSQLException: ERROR: relation "t_ds_plugin_define" does not exist
  Position: 23
; bad SQL grammar []; nested exception is org.postgresql.util.PSQLException: ERROR: relation "t_ds_plugin_define" does not exist
  Position: 23
	at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:239)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:72)
	at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:74)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:440)
	at com.sun.proxy.$Proxy11.selectList(Unknown Source)
	at org.mybatis.spring.SqlSessionTemplate.selectList(SqlSessionTemplate.java:223)
	at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.executeForMany(MybatisMapperMethod.java:158)
	at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.execute(MybatisMapperMethod.java:76)
	at com.baomidou.mybatisplus.core.override.MybatisMapperProxy.invoke(MybatisMapperProxy.java:61)
	at com.sun.proxy.$Proxy15.queryByNameAndType(Unknown Source)
	at org.apache.dolphinscheduler.dao.PluginDao.addOrUpdatePluginDefine(PluginDao.java:67)
	at org.apache.dolphinscheduler.alert.plugin.AlertPluginManager.installPlugin(AlertPluginManager.java:111)
	at org.apache.dolphinscheduler.common.plugin.DolphinPluginLoader.loadPlugin(DolphinPluginLoader.java:111)
	at org.apache.dolphinscheduler.common.plugin.DolphinPluginLoader.loadPlugin(DolphinPluginLoader.java:99)
	at org.apache.dolphinscheduler.common.plugin.DolphinPluginLoader.loadPlugins(DolphinPluginLoader.java:85)
	at org.apache.dolphinscheduler.alert.AlertServer.initPlugin(AlertServer.java:102)
	... 2 more
Caused by: org.postgresql.util.PSQLException: ERROR: relation "t_ds_plugin_define" does not exist
  Position: 23
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:143)
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:132)
	at com.alibaba.druid.pool.DruidPooledPreparedStatement.execute(DruidPooledPreparedStatement.java:497)
	at org.apache.ibatis.executor.statement.PreparedStatementHandler.query(PreparedStatementHandler.java:64)
	at org.apache.ibatis.executor.statement.RoutingStatementHandler.query(RoutingStatementHandler.java:79)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.ibatis.plugin.Plugin.invoke(Plugin.java:63)
	at com.sun.proxy.$Proxy23.query(Unknown Source)
	at com.baomidou.mybatisplus.core.executor.MybatisSimpleExecutor.doQuery(MybatisSimpleExecutor.java:67)
	at org.apache.ibatis.executor.BaseExecutor.queryFromDatabase(BaseExecutor.java:324)
	at org.apache.ibatis.executor.BaseExecutor.query(BaseExecutor.java:156)
	at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:109)
	at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:83)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:147)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:140)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:426)
	... 14 more
End start alert-server.
2021-04-26 00:27:43,637 INFO exited: alert (exit status 0; expected)
```


**Which version of Dolphin Scheduler:**
 -[dev]

**Additional context**
Add any other context about the problem here.

**Requirement or improvement**
- Please describe about your requirements or improvement suggestions.

--- a/dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/AlertServer.java
+++ b/dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/AlertServer.java
@@ -30,6 +30,7 @@ import org.apache.dolphinscheduler.common.thread.Stopper;
 import org.apache.dolphinscheduler.common.utils.PropertyUtils;
 import org.apache.dolphinscheduler.dao.AlertDao;
 import org.apache.dolphinscheduler.dao.DaoFactory;
+import org.apache.dolphinscheduler.dao.PluginDao;
 import org.apache.dolphinscheduler.dao.entity.Alert;
 import org.apache.dolphinscheduler.remote.NettyRemotingServer;
 import org.apache.dolphinscheduler.remote.command.CommandType;
@@ -47,15 +48,20 @@ import com.google.common.collect.ImmutableList;
  * alert of start
  */
 public class AlertServer {
+
     private static final Logger logger = LoggerFactory.getLogger(AlertServer.class);
+
+    /**
+     * Plugin Dao
+     */
+    private PluginDao pluginDao = DaoFactory.getDaoInstance(PluginDao.class);
+
     /**
      * Alert Dao
      */
     private AlertDao alertDao = DaoFactory.getDaoInstance(AlertDao.class);
 
     private AlertSender alertSender;
-
-    private static AlertServer instance;
 
     private AlertPluginManager alertPluginManager;
 
@@ -78,11 +84,17 @@ public class AlertServer {
 
     public static final AlertServer getInstance() {
         return AlertServerHolder.INSTANCE;
-
     }
 
     private AlertServer() {
 
+    }
+
+    private void checkTable() {
+        if (!pluginDao.checkPluginDefineTableExist()) {
+            logger.error("Plugin Define Table t_ds_plugin_define Not Exist . Please Create it First !");
+            System.exit(1);
+        }
     }
 
     private void initPlugin() {
@@ -101,7 +113,7 @@ public class AlertServer {
         try {
             alertPluginLoader.loadPlugins();
         } catch (Exception e) {
-            throw new RuntimeException("load Alert Plugin Failed !", e);
+            throw new RuntimeException("Load Alert Plugin Failed !", e);
         }
     }
 
@@ -128,7 +140,7 @@ public class AlertServer {
                 Thread.currentThread().interrupt();
             }
             if (alertPluginManager == null || alertPluginManager.getAlertChannelMap().size() == 0) {
-                logger.warn("No Alert Plugin . Can not send alert info. ");
+                logger.warn("No Alert Plugin . Cannot send alert info. ");
             } else {
                 List<Alert> alerts = alertDao.listWaitExecutionAlert();
                 alertSender = new AlertSender(alerts, alertDao, alertPluginManager);
@@ -142,6 +154,7 @@ public class AlertServer {
      */
     public void start() {
         PropertyUtils.loadPropertyFile(ALERT_PROPERTIES_PATH);
+        checkTable();
         initPlugin();
         initRemoteServer();
         logger.info("alert server ready start ");

--- a/dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/AlertServerTest.java
+++ b/dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/AlertServerTest.java
@@ -56,6 +56,7 @@ public class AlertServerTest {
 
         PluginDao pluginDao = PowerMockito.mock(PluginDao.class);
         PowerMockito.when(DaoFactory.getDaoInstance(PluginDao.class)).thenReturn(pluginDao);
+        PowerMockito.when(pluginDao.checkPluginDefineTableExist()).thenReturn(true);
 
         AlertChannel alertChannelMock = PowerMockito.mock(AlertChannel.class);
 
@@ -80,15 +81,11 @@ public class AlertServerTest {
         AlertServer alertServer = AlertServer.getInstance();
         Assert.assertNotNull(alertServer);
 
-        new Thread(() -> {
-            alertServer.start();
-        })
-                .start();
+        new Thread(() -> alertServer.start()).start();
 
         Thread.sleep(5 * Constants.ALERT_SCAN_INTERVAL);
 
         alertServer.stop();
-
     }
 
 }

--- a/dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java
+++ b/dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java
@@ -45,6 +45,15 @@ public class PluginDao extends AbstractBaseDao {
     }
 
     /**
+     * check plugin define table exist
+     *
+     * @return boolean
+     */
+    public boolean checkPluginDefineTableExist() {
+        return pluginDefineMapper.checkTableExist() > 0;
+    }
+
+    /**
      * add pluginDefine
      *
      * @param pluginDefine plugin define entiy

--- a/dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/PluginDefineMapper.java
+++ b/dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/PluginDefineMapper.java
@@ -28,6 +28,13 @@ import com.baomidou.mybatisplus.core.mapper.BaseMapper;
 public interface PluginDefineMapper extends BaseMapper<PluginDefine> {
 
     /**
+     * check table exist
+     *
+     * @return boolean
+     */
+    int checkTableExist();
+
+    /**
      * query all plugin define
      *
      * @return PluginDefine list

--- a/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/PluginDefineMapper.xml
+++ b/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/PluginDefineMapper.xml
@@ -18,6 +18,10 @@
 
 <!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd" >
 <mapper namespace="org.apache.dolphinscheduler.dao.mapper.PluginDefineMapper">
+    <select id="checkTableExist" resultType="int">
+        select count(*) from information_schema.TABLES where table_name = 't_ds_plugin_define'
+    </select>
+
     <select id="queryAllPluginDefineList" resultType="org.apache.dolphinscheduler.dao.entity.PluginDefine">
         select *
         from t_ds_plugin_define

